### PR TITLE
Fix: UTC time to active importers

### PR DESCRIPTION
**Pull request by gildub**
Add /src/app/dayjs.ts with common options
Add UTC time to importers 
Add `formatDateTime` function

Fix https://github.com/trustification/trustify-ui/issues/68

Screenshot from 2024-06-24 15-22-21

### CODE REVIEW

**Review by helio-frota — APPROVED** [review]
lgtm

File: client/src/app/pages/importer-list/importer-list.tsx [hunk 1]
```diff
@@ -250,14 +251,21 @@ export const ImporterList: React.FC = () => {
                           modifier="truncate"
                           {...getTdProps({ columnKey: "start" })}
                         >
-                          {formatDate(item.report?.startDate)}
+                          {item.state &&
+                            configValues?.disabled == false &&
+                            dayjs(item.report?.startDate)
+                              .utc()
+                              .format("YYYY-MM-DD HH:mm:ss")}
```
Comment (carlosthe19916) [inline client/src/app/pages/importer-list/importer-list.tsx:258]: Just an idea:
-  The current code in the main branch has a function `formatDate()` that can be reused everywhere.
- what about having something like `formatDateTime` so we can reuse it too?

**Review by carlosthe19916 — CHANGES_REQUESTED** [review]
Minor comments, please take a look

File: client/src/app/dayjs.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import customParseFormat from "dayjs/plugin/customParseFormat";
+import relativeTime from "dayjs/plugin/relativeTime";
+import localizedFormat from "dayjs/plugin/localizedFormat";
+
+dayjs.extend(utc);
+dayjs.extend(timezone);
+dayjs.extend(customParseFormat);
+dayjs.extend(isSameOrBefore);
+dayjs.extend(relativeTime);
+dayjs.extend(localizedFormat);
```
[the remark this answers is not in the document]
Comment (gildub) [reply client/src/app/dayjs.ts]: @carlosthe19916, yes indeed, I missed the import file, I fixed it, thanks

File: client/src/app/pages/importer-list/importer-list.tsx (hunk 1 above)
Comment (gildub) [reply client/src/app/pages/importer-list/importer-list.tsx:258]: @carlosthe19916, 

Well I suppose this a question of reuse vs performance.
I'm not against a format function here but what's do we gain as the call to dayjs format is straightforward here.

Comment (carlosthe19916) [reply client/src/app/pages/importer-list/importer-list.tsx:258]: Having formats in a single place will help us to change the formats in a single shot when needed. So far this is the only place where we use this format but in the case of `formatDate` if UX decides the format needs to change We will need to only change that function and the whole app will have the desired format rather than changing `.format(ABCAD)` in every single file

**Comment by carlosthe19916** [comment]
@helio-frota your report is not accurate. Pull the latest changes of the main branch

**Comment by helio-frota** [comment]
@carlosthe19916 that's correct sorry for the noise, the function is being used and using `dayjs`.

in this case I'm +1 having a `formatDateTime`

**Review by helio-frota — CHANGES_REQUESTED** [review]
There is a function inside `client/src/app/utils/utils.ts` that makes use of dayjs already.

export const formatDate = (value?: string) => {
  return value ? dayjs(value).format(RENDER_DATE_FORMAT) : undefined;
};

so I'm +1 to @carlosthe19916 suggestion

**Review by helio-frota — APPROVED** [review]

File: client/src/app/pages/advisory-details/vulnerabilities.tsx [hunk 3]
```diff
@@ -137,10 +130,14 @@ export const Vulnerabilities: React.FC<VulnerabilitiesProps> = ({
                       {item.non_normative.title}
                     </Td>
                     <Td width={10} {...getTdProps({ columnKey: "discovered" })}>
-                      {formatDate(item.non_normative.discovered)}
+                      {!!item.non_normative.discovered
+                        ? formatDate(item.non_normative.discovered)
+                        : null}
```
Comment (carlosthe19916) [inline client/src/app/pages/advisory-details/vulnerabilities.tsx:135]: why the unnecessary change of this block of code? what problem are we solving with this?

Comment (gildub) [reply client/src/app/pages/advisory-details/vulnerabilities.tsx:135]: @carlosthe19916, It's because I've made changes to https://github.com/trustification/trustify-ui/pull/70/files#diff-93d1e679f3a44569699332c49a4e0cf4e30ada958d50d897b3764cf2e20dd6f7R37-R39 where it makes more sense to not return `undefined` and therefore check for such case before calling `formatDate`.

Comment (carlosthe19916) [reply client/src/app/pages/advisory-details/vulnerabilities.tsx:135]: - Sorry but this change is replacing a simple line of code with a complex conditional longer code. This is is a step back in my opinion.
- What specific problem is this change solving that the previous code was not covering?

Comment (gildub) [reply client/src/app/pages/advisory-details/vulnerabilities.tsx:135]: @carlosthe19916, 

`formatDate(item.non_normative.discovered)` what happens when formatDate returns undefined ?
React might break in such case and what should be done is to trap such case and return null.

Comment (carlosthe19916) [reply client/src/app/pages/advisory-details/vulnerabilities.tsx:135]: - `formatDate` is already returning undefined in multiple cases. 
```shell
const myVar = formatDate(undefined); // Then myVar will be undefined

<>
{myVar}
</>
```

How is the previous code breaking reactjs? Can you give me a real concrete example and not suppositions?

-  I think the case is simple: Case1 vs Case2, let's explain why:

```shell
{!!item.non_normative.discovered
                        ? formatDate(item.non_normative.discovered)
                        : null}
```

is better than 

```
{formatDate(item.non_normative.released)}
```

Why if `item.non_normative.released` is undefined it will break the code?

- Couldn't we just do the following?:
```shell
export const formatDate = (value?: string) => {
  return value ? dayjs(value).format(RENDER_DATE_FORMAT) : null;
};
```

we are returning null as you want.

Comment (gildub) [reply client/src/app/pages/advisory-details/vulnerabilities.tsx:135]: @carlosthe19916 , sorry for the confusion, what I mean by breaking the code is not reactjs friendly when the code renders {undefined}.
Well it seems that React 18 handles undefined better. But that shouldn't really happen anyway : 
https://blog.saeloun.com/2022/04/14/react-18-allows-components-to-render-undfined/

Let's use your latest example and make sure formatDate and formatDateTime return null if undefined.

**Review by carlosthe19916 — APPROVED** [review]
